tests/test_kcca.py

```diff
@@ -126,4 +141,33 @@ def test_no_weights():
         kcca_b = KCCA(ktype ="linear", reg = 0.001, n_components = 1)
         kcca_b.transform([train1, train2])
         
-    
\ No newline at end of file
+# Test mrank error
+def test_mrank_neg():
+    with pytest.raises(ValueError):
+        kcca_z = KCCA(ktype ="poly", decomp = "icd", reg = 0.001, mrank = -1)
+
+# Test mrank error
+def test_mrank_large():
+    with pytest.raises(ValueError):
+        kcca_y = KCCA(ktype ="poly", decomp = "icd", reg = 0.001, mrank = 200)
+
+# Test method error
+def test_method():
+    with pytest.raises(ValueError):
+        kcca_v = KCCA(ktype ="poly", method = "test", reg = 0.001, mrank = 5)
+
+# Test precision error
+def test_precision():
+    with pytest.raises(ValueError):
+        kcca_x = KCCA(ktype ="poly", decomp = "icd", reg = 0.001, precision = -1)
+
+# Test that _zscore works
+def test_icd_mrank():
+    kcca_g_icd = KCCA(ktype ="gaussian", sigma = 1.0, n_components = 2, reg = 0.01, decomp = 'icd', mrank = 2)
+    icd = kcca_g_icd.fit_transform([x, y])
+    assert (len(icd) == 2)
+
+# Test make_icd_kernel
+def test_make_icd_kernel(): 
```

Comment:
Should test linear, gaussian, and polynomial kernel in both full and icd

Reply:
added the three kernels for icd. already did it for full.